### ISSUE TITLE
[Bug]: NullPointerException

### ISSUE THREAD

**Issue by Cha195** [Baseflow/flutter-geolocator#1291]
### Please check the following before submitting a new issue.

- [X] I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-geolocator/blob/main/geolocator/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [X] Android
- [ ] iOS
- [ ] Linux
- [ ] macOS
- [ ] Web
- [ ] Windows

### Steps to reproduce

Package throws the following error

```console
PlatformException(error, Attempt to invoke virtual method 'java.lang.String java.lang.String.trim()' on a null object reference, null, )
```

### Expected results

Run without exception

### Actual results

`Geolocator.getCurrentPosition()` throws a `NullPointerException`

### Code sample

<details><summary>Code sample</summary>

```dart
return await Geolocator.getCurrentPosition(
  desiredAccuracy: LocationAccuracy.medium,
  timeLimit: const Duration(seconds: 15),
);
```

</details>

### Version

9.0.2

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.3.1 22E772610a darwin-x64, locale en-IN)
    • Flutter version 3.10.6 on channel stable at /Users/sricharanramesh/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (2 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/sricharanramesh/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] IntelliJ IDEA Ultimate Edition (version 2023.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 73.1.1
    • Dart plugin version 231.9065

[✓] VS Code (version 1.80.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.68.0

[✓] Connected device (3 available)
    • SM E625F (mobile) • RZ8R22H0VPJ • android-arm64  • Android 13 (API 33)
    • macOS (desktop)   • macos       • darwin-x64     • macOS 13.3.1 22E772610a darwin-x64
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 113.0.5672.126

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by JeroenWeener** [comment]
Hi @Cha195, thank you for reaching out.

Could you check if you're still experiencing this error on the latest version of geolocator (10.0.1)?
It seems that the plugin could not find an appropriate location provider. The logic for that has been updated recently.

**Comment by Cha195** [comment]
Hi @JeroenWeener. Sorry for the really late reply. It seems the crash has been resolved post update to 10.0.0. Thanks for the help!

**Comment by JeroenWeener** [comment]
@Cha195 no problem, thanks for verifying!
